app/models/membership.rb

```diff
@@ -89,6 +89,6 @@ def discussion_readers
   private
 
   def set_volume
-    self.volume = user.default_membership_volume if group.is_formal_group?
+    self.volume = user.default_membership_volume if id.nil? && group.is_formal_group?
```

Comment:
Oh hmm what is this change needed for?

I think more idiomatic is `self.new_record?`

Reply:
I've working to remove all unnecessary N+1 queries. Reading volume triggers a query in a couple of places (checking default membership volume and group volume) and it also pollutes the export because it returns a value that isn't actually what the column actually contains.

So that's why I've moved to using a method that clearly says it's giving a computed value for volume rather than overloading the simple accessor. I like it more and it hasn't really been a problem to change.